unit flake: TestRequestAttachRemoteCommand

http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/5941/

```
=== RUN   TestRequestAttachRemoteCommand
panic: test timed out after 5m0s

goroutine 207 [running]:
testing.startAlarm.func1()
	/tmp/workdir/go/src/testing/testing.go:703 +0x16b
created by time.goFunc
	/tmp/workdir/go/src/time/sleep.go:129 +0x6e
```

cc @kubernetes/goog-cluster @krousey

```
panic: test timed out after 5m0s

goroutine 207 [running]:
testing.startAlarm.func1()
	/tmp/workdir/go/src/testing/testing.go:703 +0x16b
created by time.goFunc
	/tmp/workdir/go/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0x1800548, 0x1e41580, 0x3, 0x3, 0x53ae01)
	/tmp/workdir/go/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820065f10, 0xc820010180)
	/tmp/workdir/go/src/testing/testing.go:494 +0xe5
main.main()
	k8s.io/kubernetes/pkg/client/unversioned/remotecommand/_test/_testmain.go:58 +0x210

goroutine 17 [syscall, 4 minutes, locked to thread]:
runtime.goexit()
	/tmp/workdir/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive, 2 minutes]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1e62740)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x8c
created by github.com/golang/glog.init.1
	/workspace/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2ba

goroutine 96 [semacquire, 4 minutes]:
sync.runtime_Semacquire(0xc82027c60c)
	/tmp/workdir/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc82027c600)
	/tmp/workdir/go/src/sync/waitgroup.go:126 +0x118
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream(0xc8206ac380, 0x7f549974e748, 0xc8201dc510, 0x0, 0x0)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:162 +0x695
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamExecutor).Stream(0xc82066e3c0, 0x7f549974d368, 0xc82069c9e0, 0x7f549974e6c0, 0xc8201fa460, 0x7f549974e6c0, 0xc8201fa540, 0x0, 0x0, 0x0)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/remotecommand.go:179 +0x43a
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.TestRequestAttachRemoteCommand(0xc8201da000)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/remotecommand_test.go:286 +0x8c5
testing.tRunner(0xc8201da000, 0x1e41598)
	/tmp/workdir/go/src/testing/testing.go:456 +0xdd
created by testing.RunTests
	/tmp/workdir/go/src/testing/testing.go:561 +0xaa4

goroutine 246 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be000, 0xc820686150)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 224 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc82066e350)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82066e340)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc8201dc390, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be210, 0xc8201dc390, 0xc82022c010)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82020a000, 0xc8206be210, 0xc82022c010, 0xc8201dc390)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 244 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be000, 0xc820686120)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 243 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc820014c50)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc820014c40)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc8206860c0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be000, 0xc8206860c0, 0xc82022c550)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82023c190, 0xc8206be000, 0xc82022c550, 0xc8206860c0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 15 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be000, 0xc820686090)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 77 [select, 4 minutes]:
github.com/docker/spdystream.(*idleAwareFramer).monitor(0xc8201dc3f0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:57 +0x662
created by github.com/docker/spdystream.NewConnection
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:227 +0x80a

goroutine 225 [select, 4 minutes]:
github.com/docker/spdystream.(*Connection).shutdown(0xc8206be210, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:678 +0x44b
created by github.com/docker/spdystream.(*Connection).Close
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:732 +0x266

goroutine 200 [select, 4 minutes]:
github.com/docker/spdystream.(*Connection).shutdown(0xc8206be000, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:678 +0x44b
created by github.com/docker/spdystream.(*Connection).handleGoAwayFrame
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:600 +0x24f

goroutine 201 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8206ac1d0)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8206ac1c0)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*Connection).shutdown.func1(0xc8206be000, 0xc8206cad80)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:671 +0x296
created by github.com/docker/spdystream.(*Connection).shutdown
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:675 +0x169

goroutine 242 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be000, 0xc8206860c0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 218 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc82066e290)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82066e280)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc8201dc270, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be210, 0xc8201dc270, 0xc82022c010)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82020a000, 0xc8206be210, 0xc82022c010, 0xc8201dc270)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 16 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc820014c10)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc820014c00)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc820686090, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be000, 0xc820686090, 0xc82022c550)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82023c190, 0xc8206be000, 0xc82022c550, 0xc820686090)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 14 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc820014bd0)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc820014bc0)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc820686060, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be000, 0xc820686060, 0xc82022c550)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82023c190, 0xc8206be000, 0xc82022c550, 0xc820686060)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 13 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be000, 0xc820686060)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 221 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be210, 0xc8201dc360)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 220 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc82066e2d0)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82066e2c0)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc8201dc300, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be210, 0xc8201dc300, 0xc82022c010)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82020a000, 0xc8206be210, 0xc82022c010, 0xc8201dc300)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 108 [chan send, 4 minutes]:
github.com/docker/spdystream.(*Connection).shutdown(0xc8202c2160, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:700 +0x335
created by github.com/docker/spdystream.(*Connection).Close
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:732 +0x266

goroutine 245 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc820014c90)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc820014c80)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc820686120, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be000, 0xc820686120, 0xc82022c550)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82023c190, 0xc8206be000, 0xc82022c550, 0xc820686120)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 247 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc820014cd0)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc820014cc0)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc820686150, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be000, 0xc820686150, 0xc82022c550)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82023c190, 0xc8206be000, 0xc82022c550, 0xc820686150)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 97 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f549974e280, 0x72, 0x7f549bfc6050)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201fa840, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201fa840, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc8201fa7e0, 0x0, 0x7f5498ec6000, 0xc82069cba0)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc82069e000, 0xc82066e440, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
net.(*TCPListener).Accept(0xc82069e000, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:264 +0x4b
net/http/httptest.(*historyListener).Accept(0xc82000b710, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/httptest/server.go:48 +0x85
net/http.(*Server).Serve(0xc82018a300, 0x7f549974e580, 0xc82000b710, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/server.go:1887 +0xc4
created by net/http/httptest.(*Server).Start
	/tmp/workdir/go/src/net/http/httptest/server.go:109 +0x4fb

goroutine 217 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be210, 0xc8201dc270)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 216 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc82066e050)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82066e040)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc8201dc210, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be210, 0xc8201dc210, 0xc82022c010)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82020a000, 0xc8206be210, 0xc82022c010, 0xc8201dc210)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 78 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f549974e400, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201faed0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201faed0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc8201fae70, 0xc82023c250, 0x4, 0x8, 0x0, 0x7f549bfc6050, 0xc820012150)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc82069e018, 0xc82023c250, 0x4, 0x8, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
net.(*TCPConn).Read(0xc82069e018, 0xc82023c250, 0x4, 0x8, 0xc820166480, 0x0, 0x0)
	<autogenerated>:74 +0x7d
io.ReadAtLeast(0x7f5498ec6050, 0xc82069e018, 0xc82023c250, 0x4, 0x8, 0x4, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:298 +0x119
io.ReadFull(0x7f5498ec6050, 0xc82069e018, 0xc82023c250, 0x4, 0x8, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:316 +0x77
encoding/binary.Read(0x7f5498ec6050, 0xc82069e018, 0x7f549974e800, 0x280dad0, 0x11b4dc0, 0xc82023c240, 0x0, 0x0)
	/tmp/workdir/go/src/encoding/binary/binary.go:148 +0x159
github.com/docker/spdystream/spdy.(*Framer).ReadFrame(0xc820472240, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/spdy/read.go:148 +0x137
github.com/docker/spdystream.(*idleAwareFramer).ReadFrame(0xc8201dc3f0, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:137 +0x5c
github.com/docker/spdystream.(*Connection).Serve(0xc8206be000, 0xc82022c550)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:299 +0x2f6
created by k8s.io/kubernetes/pkg/util/httpstream/spdy.newConnection
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/httpstream/spdy/connection.go:68 +0x17d

goroutine 215 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be210, 0xc8201dc210)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 222 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc82066e310)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82066e300)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc8201dc360, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc8206be210, 0xc8201dc360, 0xc82022c010)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc82020a000, 0xc8206be210, 0xc82022c010, 0xc8201dc360)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 223 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be210, 0xc8201dc390)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 219 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc8206be210, 0xc8201dc300)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 258 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8206ac090)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8206ac080)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*Connection).shutdown.func1(0xc8206be210, 0xc82018a240)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:671 +0x296
created by github.com/docker/spdystream.(*Connection).shutdown
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:675 +0x169

goroutine 248 [select, 4 minutes]:
github.com/docker/spdystream.(*idleAwareFramer).monitor(0xc82000a600)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:57 +0x662
created by github.com/docker/spdystream.NewConnection
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:227 +0x80a

goroutine 249 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f549974e4c0, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201fafb0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201fafb0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc8201faf50, 0xc8201640a0, 0x4, 0x8, 0x0, 0x7f549bfc6050, 0xc820012150)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc82069e028, 0xc8201640a0, 0x4, 0x8, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
net.(*TCPConn).Read(0xc82069e028, 0xc8201640a0, 0x4, 0x8, 0xc820638480, 0x0, 0x0)
	<autogenerated>:74 +0x7d
io.ReadAtLeast(0x7f5498ec6050, 0xc82069e028, 0xc8201640a0, 0x4, 0x8, 0x4, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:298 +0x119
io.ReadFull(0x7f5498ec6050, 0xc82069e028, 0xc8201640a0, 0x4, 0x8, 0x4, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:316 +0x77
encoding/binary.Read(0x7f5498ec6050, 0xc82069e028, 0x7f549974e800, 0x280dad0, 0x11b4dc0, 0xc820164090, 0x0, 0x0)
	/tmp/workdir/go/src/encoding/binary/binary.go:148 +0x159
github.com/docker/spdystream/spdy.(*Framer).ReadFrame(0xc8204720c0, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/spdy/read.go:148 +0x137
github.com/docker/spdystream.(*idleAwareFramer).ReadFrame(0xc82000a600, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:137 +0x5c
github.com/docker/spdystream.(*Connection).Serve(0xc8206be210, 0xc82022c010)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:299 +0x2f6
created by k8s.io/kubernetes/pkg/util/httpstream/spdy.newConnection
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/httpstream/spdy/connection.go:68 +0x17d

goroutine 202 [chan send, 4 minutes]:
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream.func1(0x7f5498ec6428, 0xc8200160a0, 0xc820472420)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:97 +0x298
created by k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:102 +0x479

goroutine 206 [select, 4 minutes]:
github.com/docker/spdystream.(*Stream).Read(0xc8204863c0, 0xc820198000, 0x200, 0x200, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/stream.go:80 +0x42f
bytes.(*Buffer).ReadFrom(0xc8201fa540, 0x7f5498ec6678, 0xc8204863c0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/bytes/buffer.go:173 +0x415
io.copyBuffer(0x7f549974e6c0, 0xc8201fa540, 0x7f5498ec6678, 0xc8204863c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:375 +0x1a6
io.Copy(0x7f549974e6c0, 0xc8201fa540, 0x7f5498ec6678, 0xc8204863c0, 0xc8204863c0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:351 +0x79
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream.func5(0xc82027c600, 0xc8206ac380, 0x7f5498ec6428, 0xc8204863c0)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:155 +0xc3
created by k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:158 +0x687
FAIL	k8s.io/kubernetes/pkg/client/unversioned/remotecommand	300.137s
```

Happened again:

```
=== RUN   TestRequestAttachRemoteCommand
panic: test timed out after 5m0s

goroutine 97 [running]:
testing.startAlarm.func1()
	/tmp/workdir/go/src/testing/testing.go:703 +0x16b
created by time.goFunc
	/tmp/workdir/go/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0x1878d30, 0x1ed66a0, 0x3, 0x3, 0x53ae01)
	/tmp/workdir/go/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc8200b5f10, 0xc8200101e0)
	/tmp/workdir/go/src/testing/testing.go:494 +0xe5
main.main()
	k8s.io/kubernetes/pkg/client/unversioned/remotecommand/_test/_testmain.go:58 +0x210

goroutine 17 [syscall, 4 minutes, locked to thread]:
runtime.goexit()
	/tmp/workdir/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive, 2 minutes]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1ef7980)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x8c
created by github.com/golang/glog.init.1
	/workspace/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2ba

goroutine 22 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fad943558c0, 0x72, 0x7fad96c0e050)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc820227560, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc820227560, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc820227500, 0x0, 0x7fad94355c10, 0xc8200cd080)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc8200d0040, 0xc8200ce840, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
net.(*TCPListener).Accept(0xc8200d0040, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:264 +0x4b
net/http/httptest.(*historyListener).Accept(0xc82025fcb0, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/httptest/server.go:48 +0x85
net/http.(*Server).Serve(0xc8200cbd40, 0x7fad94355980, 0xc82025fcb0, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/server.go:1887 +0xc4
created by net/http/httptest.(*Server).Start
	/tmp/workdir/go/src/net/http/httptest/server.go:109 +0x4fb

goroutine 258 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc820294160, 0xc8202aa060)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 15 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fad94355740, 0x72, 0x7fad96c0e050)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc820170290, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc820170290, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc820170230, 0x0, 0x7fad94355c10, 0xc8205f6580)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc82014e018, 0xc8203cc500, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
net.(*TCPListener).Accept(0xc82014e018, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:264 +0x4b
net/http/httptest.(*historyListener).Accept(0xc82025eed0, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/httptest/server.go:48 +0x85
net/http.(*Server).Serve(0xc8200ca120, 0x7fad94355980, 0xc82025eed0, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/server.go:1887 +0xc4
created by net/http/httptest.(*Server).Start
	/tmp/workdir/go/src/net/http/httptest/server.go:109 +0x4fb

goroutine 242 [select, 4 minutes]:
github.com/docker/spdystream.(*idleAwareFramer).monitor(0xc820208120)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:57 +0x662
created by github.com/docker/spdystream.NewConnection
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:227 +0x80a

goroutine 184 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc82010c420, 0xc820162300)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 185 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8200ce790)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8200ce780)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc820162300, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc82010c420, 0xc820162300, 0xc820013bf0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820013c00, 0xc82010c420, 0xc820013bf0, 0xc820162300)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 147 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fad94355800, 0x72, 0x7fad96c0e050)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc820018290, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc820018290, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc820018230, 0x0, 0x7fad94355c10, 0xc8200ccc60)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc8200d0000, 0xc8200ce5c0, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
net.(*TCPListener).Accept(0xc8200d0000, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:264 +0x4b
net/http/httptest.(*historyListener).Accept(0xc820162120, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/httptest/server.go:48 +0x85
net/http.(*Server).Serve(0xc8201fc360, 0x7fad94355980, 0xc820162120, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/server.go:1887 +0xc4
created by net/http/httptest.(*Server).Start
	/tmp/workdir/go/src/net/http/httptest/server.go:109 +0x4fb

goroutine 113 [semacquire, 4 minutes]:
sync.runtime_Semacquire(0xc82014497c)
	/tmp/workdir/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc820144970)
	/tmp/workdir/go/src/sync/waitgroup.go:126 +0x118
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream(0xc8201dc100, 0x7fad94355e70, 0xc8202081b0, 0x0, 0x0)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:162 +0x695
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamExecutor).Stream(0xc820140bc0, 0x7fad94354850, 0xc82000ed80, 0x7fad94355ac0, 0xc820158000, 0x7fad94355ac0, 0xc820158070, 0x0, 0x0, 0x0)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/remotecommand.go:179 +0x43a
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.TestRequestAttachRemoteCommand(0xc8200a4000)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/remotecommand_test.go:289 +0x8d1
testing.tRunner(0xc8200a4000, 0x1ed66b8)
	/tmp/workdir/go/src/testing/testing.go:456 +0xdd
created by testing.RunTests
	/tmp/workdir/go/src/testing/testing.go:561 +0xaa4

goroutine 226 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fad94355500, 0x72, 0x7fad96c0e050)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201581b0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201581b0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc820158150, 0x0, 0x7fad94355c10, 0xc82000ee80)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc8200d0018, 0xc820140c00, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
net.(*TCPListener).Accept(0xc8200d0018, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:264 +0x4b
net/http/httptest.(*historyListener).Accept(0xc82000b260, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/httptest/server.go:48 +0x85
net/http.(*Server).Serve(0xc8202ac6c0, 0x7fad94355980, 0xc82000b260, 0x0, 0x0)
	/tmp/workdir/go/src/net/http/server.go:1887 +0xc4
created by net/http/httptest.(*Server).Start
	/tmp/workdir/go/src/net/http/httptest/server.go:109 +0x4fb

goroutine 231 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc820140c90)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc820140c80)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*Connection).shutdown.func1(0xc82010c420, 0xc8200ca420)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:671 +0x296
created by github.com/docker/spdystream.(*Connection).shutdown
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:675 +0x169

goroutine 228 [select, 4 minutes]:
github.com/docker/spdystream.(*idleAwareFramer).monitor(0xc82000b740)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:57 +0x662
created by github.com/docker/spdystream.NewConnection
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:227 +0x80a

goroutine 229 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fad94355680, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8201583e0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8201583e0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc820158380, 0xc820144a70, 0x4, 0x8, 0x0, 0x7fad96c0e050, 0xc820012150)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc8200d0048, 0xc820144a70, 0x4, 0x8, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
net.(*TCPConn).Read(0xc8200d0048, 0xc820144a70, 0x4, 0x8, 0xc820134900, 0x0, 0x0)
	<autogenerated>:74 +0x7d
io.ReadAtLeast(0x7fad94355b70, 0xc8200d0048, 0xc820144a70, 0x4, 0x8, 0x4, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:298 +0x119
io.ReadFull(0x7fad94355b70, 0xc8200d0048, 0xc820144a70, 0x4, 0x8, 0x4, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:316 +0x77
encoding/binary.Read(0x7fad94355b70, 0xc8200d0048, 0x7fad94355f28, 0x28a2d68, 0x12115a0, 0xc820144a60, 0x0, 0x0)
	/tmp/workdir/go/src/encoding/binary/binary.go:148 +0x159
github.com/docker/spdystream/spdy.(*Framer).ReadFrame(0xc8202ac900, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/spdy/read.go:148 +0x137
github.com/docker/spdystream.(*idleAwareFramer).ReadFrame(0xc82000b740, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:137 +0x5c
github.com/docker/spdystream.(*Connection).Serve(0xc82010c420, 0xc820013bf0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:299 +0x2f6
created by k8s.io/kubernetes/pkg/util/httpstream/spdy.newConnection
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/httpstream/spdy/connection.go:68 +0x17d

goroutine 243 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fad943555c0, 0x72, 0x0)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc820158370, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc820158370, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).Read(0xc820158310, 0xc820144950, 0x4, 0x8, 0x0, 0x7fad96c0e050, 0xc820012150)
	/tmp/workdir/go/src/net/fd_unix.go:232 +0x27b
net.(*conn).Read(0xc82014e020, 0xc820144950, 0x4, 0x8, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/net/net.go:172 +0x121
net.(*TCPConn).Read(0xc82014e020, 0xc820144950, 0x4, 0x8, 0xc820134900, 0x0, 0x0)
	<autogenerated>:74 +0x7d
io.ReadAtLeast(0x7fad94355b70, 0xc82014e020, 0xc820144950, 0x4, 0x8, 0x4, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:298 +0x119
io.ReadFull(0x7fad94355b70, 0xc82014e020, 0xc820144950, 0x4, 0x8, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:316 +0x77
encoding/binary.Read(0x7fad94355b70, 0xc82014e020, 0x7fad94355f28, 0x28a2d68, 0x12115a0, 0xc820144940, 0x0, 0x0)
	/tmp/workdir/go/src/encoding/binary/binary.go:148 +0x159
github.com/docker/spdystream/spdy.(*Framer).ReadFrame(0xc8201680c0, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/spdy/read.go:148 +0x137
github.com/docker/spdystream.(*idleAwareFramer).ReadFrame(0xc820208120, 0x0, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:137 +0x5c
github.com/docker/spdystream.(*Connection).Serve(0xc820294160, 0xc82014a1e0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:299 +0x2f6
created by k8s.io/kubernetes/pkg/util/httpstream/spdy.newConnection
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/httpstream/spdy/connection.go:68 +0x17d

goroutine 259 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8202a8090)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8202a8080)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc8202aa060, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc820294160, 0xc8202aa060, 0xc82014a1e0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820144100, 0xc820294160, 0xc82014a1e0, 0xc8202aa060)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 186 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc820294160, 0xc82000a630)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 187 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc1d0)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc1c0)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000a630, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc820294160, 0xc82000a630, 0xc82014a1e0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820144100, 0xc820294160, 0xc82014a1e0, 0xc82000a630)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 188 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc820294160, 0xc82000a750)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 189 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc210)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc200)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000a750, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc820294160, 0xc82000a750, 0xc82014a1e0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820144100, 0xc820294160, 0xc82014a1e0, 0xc82000a750)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 190 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc820294160, 0xc82000ade0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 191 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc250)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc240)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000ade0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc820294160, 0xc82000ade0, 0xc82014a1e0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820144100, 0xc820294160, 0xc82014a1e0, 0xc82000ade0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 192 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc820294160, 0xc82000ae10)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 193 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc290)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc280)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000ae10, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc820294160, 0xc82000ae10, 0xc82014a1e0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820144100, 0xc820294160, 0xc82014a1e0, 0xc82000ae10)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 274 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc82010c420, 0xc82000a600)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 275 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc090)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc080)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000a600, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc82010c420, 0xc82000a600, 0xc820013bf0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820013c00, 0xc82010c420, 0xc820013bf0, 0xc82000a600)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 276 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc82010c420, 0xc82000b140)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 277 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc510)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc500)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000b140, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc82010c420, 0xc82000b140, 0xc820013bf0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820013c00, 0xc82010c420, 0xc820013bf0, 0xc82000b140)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 278 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc82010c420, 0xc82000b1a0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 279 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc550)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc540)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000b1a0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc82010c420, 0xc82000b1a0, 0xc820013bf0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820013c00, 0xc82010c420, 0xc820013bf0, 0xc82000b1a0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 280 [chan receive, 4 minutes]:
github.com/docker/spdystream.(*Connection).Serve.func1(0xc82010c420, 0xc82000b2c0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:281 +0x5b
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:283 +0x1f4

goroutine 281 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc590)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc580)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*PriorityFrameQueue).Pop(0xc82000b2c0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/priority.go:86 +0x1e4
github.com/docker/spdystream.(*Connection).frameHandler(0xc82010c420, 0xc82000b2c0, 0xc820013bf0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:372 +0x5a
github.com/docker/spdystream.(*Connection).Serve.func2(0xc820013c00, 0xc82010c420, 0xc820013bf0, 0xc82000b2c0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:290 +0x6a
created by github.com/docker/spdystream.(*Connection).Serve
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:291 +0x2a2

goroutine 230 [select, 4 minutes]:
github.com/docker/spdystream.(*Connection).shutdown(0xc82010c420, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:678 +0x44b
created by github.com/docker/spdystream.(*Connection).Close
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:732 +0x266

goroutine 232 [select, 4 minutes]:
github.com/docker/spdystream.(*Connection).shutdown(0xc820294160, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:678 +0x44b
created by github.com/docker/spdystream.(*Connection).handleGoAwayFrame
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:600 +0x24f

goroutine 233 [chan send, 4 minutes]:
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream.func1(0x7fad96c138c0, 0xc8205f0000, 0xc8201682a0)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:97 +0x298
created by k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:102 +0x479

goroutine 238 [semacquire, 4 minutes]:
sync.runtime_Syncsemacquire(0xc8201dc0d0)
	/tmp/workdir/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8201dc0c0)
	/tmp/workdir/go/src/sync/cond.go:62 +0xb3
github.com/docker/spdystream.(*Connection).shutdown.func1(0xc820294160, 0xc8200ca480)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:671 +0x296
created by github.com/docker/spdystream.(*Connection).shutdown
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:675 +0x169

goroutine 237 [select, 4 minutes]:
github.com/docker/spdystream.(*Stream).Read(0xc8205f05a0, 0xc820174400, 0x200, 0x200, 0x0, 0x0, 0x0)
	/workspace/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/stream.go:80 +0x42f
bytes.(*Buffer).ReadFrom(0xc820158070, 0x7fad943564d8, 0xc8205f05a0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/bytes/buffer.go:173 +0x415
io.copyBuffer(0x7fad94355ac0, 0xc820158070, 0x7fad943564d8, 0xc8205f05a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:375 +0x1a6
io.Copy(0x7fad94355ac0, 0xc820158070, 0x7fad943564d8, 0xc8205f05a0, 0xc8205f05a0, 0x0, 0x0)
	/tmp/workdir/go/src/io/io.go:351 +0x79
k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream.func5(0xc820144970, 0xc8201dc100, 0x7fad96c138c0, 0xc8205f05a0)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:155 +0xc3
created by k8s.io/kubernetes/pkg/client/unversioned/remotecommand.(*streamProtocolV2).stream
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/unversioned/remotecommand/v2.go:158 +0x687
FAIL	k8s.io/kubernetes/pkg/client/unversioned/remotecommand	300.161s
```

https://00e9e64bac644d7913fff76cec9b8d18d3b9bbce76c45efb91-apidata.googleusercontent.com/download/storage/v1_internal/b/kubernetes-jenkins/o/pr-logs%2F23fd5e9a33c5ab42391deb7704f23166128d3ca4%2Fkubernetes-pull-test-unit-integration%2F10413%2Fartifacts%2Fjunit_v1,extensions-v1beta1_20160122-005557.stdout?qk=AD5uMEsBDT24OsDI58nAhrjTbRfLhl27XbWPTiMTCzs3kiVk4NUtcJDTK9_WAdhJeeSvDxM9yeItHiEAoIgj1Iu5d82ArsPbwRAsA_9mmToA3DUbeS8psHVC17W3GkElu7T1fmaeNHS-VV0m-I4pbiq8rYprJ0oC82L-8rrLIsO2faZljL2nTR6urIqntP9yTPhcHmEdZdWWbq7LZ2xHLoyXgXbGYOX0Mx2nzjZus60otFPOyz22B_69m7gfSFM3YLK-XM7cmHO62VIE4M_q9Xj17UvXwS0F0b9m3tr2o3x24I1w0F46SPauGDmSDBCaKYDz4drcLmP5k-WAGCx5XMcE5KNYTo2owrENzWoogfCBtRjyzkIVnpON_0RNAvLMjCKbeKni1sBtkmlHpXF16VOaI0_Wqaet33WkiYvRmxjhfOZ6MBWL_73jQP7QmVWLVCJYH73C0jF0Ua_Jm9K3URY8kmQ0Qd0lkFP2XwK0rCh4kLPGvikILabE13kT01mc6NLcq6MVbwrhZ7SlX3_7HPKMPDx0Avar_mtiR9TMSJhny-1WZgGrJwgRDNqEiNIvf6VeNMgFK_CKCn3z08RRS6GkfWvQTGz0wsBeDrewuZ1rBhN1mECrGYf9ww-TRMi285BrDSDSfHSXP_Fq5BQoHdZWbnUMksQr7HzBQlGfmcpZjyqGV56j7n-oQgtXDHmWDGjymd1_vyvjnN0SPqhXP_jCNfDfapSVL8QCwEaCiUwwMUevI3W6XW8M99VnERQMwBlFDanosajb0dypi3icvSJj1q_DnW2ZD-kKUV_uFTmMNAjiOkgxEKL4SfosTjFisKthHZp61Ij1Y__Dm-BKo2a28r_PmQtosCDeiw12aeOWiDyXG5sP5tLRDaSoIXG1Hez8sWUcdGVEl9Qtj0UdU2yVFh-liHbR6J3uB89zmZKqSl6F_ByRY4WsrXEZMvaeB5uwa2RvBM0UZ2IJx3q4hDsY80yCPUreZKVWwJ32CS-cxf2yeq55dvajDNtXRE0Elq7ojicxVUJawiB-D_vUSOXh3EFjK_6M9g

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -403,7 +403,7 @@
 		},
 		{
 			"ImportPath": "github.com/docker/spdystream",
-			"Rev": "c33989bcb56748d2473194d11f8ac3fc563688eb"
+			"Rev": "106e140db2cb50923efe088bf2906b2ee5a45fec"
 		},
 		{
 			"ImportPath": "github.com/elazarl/go-bindata-assetfs",

--- a/Godeps/_workspace/src/github.com/docker/spdystream/connection.go
+++ b/Godeps/_workspace/src/github.com/docker/spdystream/connection.go
@@ -320,6 +320,7 @@ func (s *Connection) Serve(newHandler StreamHandler) {
 		partitionRoundRobin int
 		goAwayFrame         *spdy.GoAwayFrame
 	)
+Loop:
 	for {
 		readFrame, err := s.framer.ReadFrame()
 		if err != nil {
@@ -362,7 +363,7 @@ func (s *Connection) Serve(newHandler StreamHandler) {
 		case *spdy.GoAwayFrame:
 			// hold on to the go away frame and exit the loop
 			goAwayFrame = frame
-			break
+			break Loop
 		default:
 			priority = 7
 			partition = partitionRoundRobin

--- a/pkg/client/unversioned/remotecommand/remotecommand_test.go
+++ b/pkg/client/unversioned/remotecommand/remotecommand_test.go
@@ -34,6 +34,19 @@ import (
 	"k8s.io/kubernetes/pkg/util/httpstream/spdy"
 )
 
+type streamAndReply struct {
+	httpstream.Stream
+	replySent <-chan struct{}
+}
+
+func waitStreamReply(replySent <-chan struct{}, notify chan<- struct{}, stop <-chan struct{}) {
+	select {
+	case <-replySent:
+		notify <- struct{}{}
+	case <-stop:
+	}
+}
+
 func fakeExecServer(t *testing.T, i int, stdinData, stdoutData, stderrData, errorData string, tty bool, messageCount int) http.HandlerFunc {
 	// error + stdin + stdout
 	expectedStreams := 3
@@ -50,11 +63,11 @@ func fakeExecServer(t *testing.T, i int, stdinData, stdoutData, stderrData, erro
 		if protocol != StreamProtocolV2Name {
 			t.Fatalf("unexpected protocol: %s", protocol)
 		}
-		streamCh := make(chan httpstream.Stream)
+		streamCh := make(chan streamAndReply)
 
 		upgrader := spdy.NewResponseUpgrader()
-		conn := upgrader.UpgradeResponse(w, req, func(stream httpstream.Stream) error {
-			streamCh <- stream
+		conn := upgrader.UpgradeResponse(w, req, func(stream httpstream.Stream, replySent <-chan struct{}) error {
+			streamCh <- streamAndReply{Stream: stream, replySent: replySent}
 			return nil
 		})
 		// from this point on, we can no longer call methods on w
@@ -68,6 +81,9 @@ func fakeExecServer(t *testing.T, i int, stdinData, stdoutData, stderrData, erro
 
 		var errorStream, stdinStream, stdoutStream, stderrStream httpstream.Stream
 		receivedStreams := 0
+		replyChan := make(chan struct{})
+		stop := make(chan struct{})
+		defer close(stop)
 	WaitForStreams:
 		for {
 			select {
@@ -76,20 +92,25 @@ func fakeExecServer(t *testing.T, i int, stdinData, stdoutData, stderrData, erro
 				switch streamType {
 				case api.StreamTypeError:
 					errorStream = stream
-					receivedStreams++
+					go waitStreamReply(stream.replySent, replyChan, stop)
 				case api.StreamTypeStdin:
 					stdinStream = stream
-					receivedStreams++
+					go waitStreamReply(stream.replySent, replyChan, stop)
 				case api.StreamTypeStdout:
 					stdoutStream = stream
-					receivedStreams++
+					go waitStreamReply(stream.replySent, replyChan, stop)
 				case api.StreamTypeStderr:
 					stderrStream = stream
-					receivedStreams++
+					go waitStreamReply(stream.replySent, replyChan, stop)
 				default:
 					t.Errorf("%d: unexpected stream type: %q", i, streamType)
 				}
 
+				if receivedStreams == expectedStreams {
+					break WaitForStreams
+				}
+			case <-replyChan:
+				receivedStreams++
 				if receivedStreams == expectedStreams {
 					break WaitForStreams
 				}

--- a/pkg/kubelet/server/server.go
+++ b/pkg/kubelet/server/server.go
@@ -606,6 +606,15 @@ func standardShellChannels(stdin, stdout, stderr bool) []wsstream.ChannelType {
 	return channels
 }
 
+// streamAndReply holds both a Stream and a channel that is closed when the stream's reply frame is
+// enqueued. Consumers can wait for replySent to be closed prior to proceeding, to ensure that the
+// replyFrame is enqueued before the connection's goaway frame is sent (e.g. if a stream was
+// received and right after, the connection gets closed).
+type streamAndReply struct {
+	httpstream.Stream
+	replySent <-chan struct{}
+}
+
 func (s *Server) createStreams(request *restful.Request, response *restful.Response) (io.Reader, io.WriteCloser, io.WriteCloser, io.WriteCloser, Closer, bool, bool) {
 	tty := request.QueryParameter(api.ExecTTYParam) == "1"
 	stdin := request.QueryParameter(api.ExecStdinParam) == "1"
@@ -664,11 +673,11 @@ func (s *Server) createStreams(request *restful.Request, response *restful.Respo
 		return nil, nil, nil, nil, nil, false, false
 	}
 
-	streamCh := make(chan httpstream.Stream)
+	streamCh := make(chan streamAndReply)
 
 	upgrader := spdy.NewResponseUpgrader()
-	conn := upgrader.UpgradeResponse(response.ResponseWriter, request.Request, func(stream httpstream.Stream) error {
-		streamCh <- stream
+	conn := upgrader.UpgradeResponse(response.ResponseWriter, request.Request, func(stream httpstream.Stream, replySent <-chan struct{}) error {
+		streamCh <- streamAndReply{Stream: stream, replySent: replySent}
 		return nil
 	})
 	// from this point on, we can no longer call methods on response
@@ -686,6 +695,9 @@ func (s *Server) createStreams(request *restful.Request, response *restful.Respo
 
 	var errorStream, stdinStream, stdoutStream, stderrStream httpstream.Stream
 	receivedStreams := 0
+	replyChan := make(chan struct{})
+	stop := make(chan struct{})
+	defer close(stop)
 WaitForStreams:
 	for {
 		select {
@@ -694,19 +706,21 @@ WaitForStreams:
 			switch streamType {
 			case api.StreamTypeError:
 				errorStream = stream
-				receivedStreams++
+				go waitStreamReply(stream.replySent, replyChan, stop)
 			case api.StreamTypeStdin:
 				stdinStream = stream
-				receivedStreams++
+				go waitStreamReply(stream.replySent, replyChan, stop)
 			case api.StreamTypeStdout:
 				stdoutStream = stream
-				receivedStreams++
+				go waitStreamReply(stream.replySent, replyChan, stop)
 			case api.StreamTypeStderr:
 				stderrStream = stream
-				receivedStreams++
+				go waitStreamReply(stream.replySent, replyChan, stop)
 			default:
 				glog.Errorf("Unexpected stream type: '%s'", streamType)
 			}
+		case <-replyChan:
+			receivedStreams++
 			if receivedStreams == expectedStreams {
 				break WaitForStreams
 			}
@@ -719,6 +733,16 @@ WaitForStreams:
 	}
 
 	return stdinStream, stdoutStream, stderrStream, errorStream, conn, tty, true
+}
+
+// waitStreamReply waits until either replySent or stop is closed. If replySent is closed, it sends
+// an empty struct to the notify channel.
+func waitStreamReply(replySent <-chan struct{}, notify chan<- struct{}, stop <-chan struct{}) {
+	select {
+	case <-replySent:
+		notify <- struct{}{}
+	case <-stop:
+	}
 }
 
 func getPodCoordinates(request *restful.Request) (namespace, pod string, uid types.UID) {
@@ -796,8 +820,8 @@ func ServePortForward(w http.ResponseWriter, req *http.Request, portForwarder Po
 // forward streams. It checks each stream's port and stream type headers,
 // rejecting any streams that with missing or invalid values. Each valid
 // stream is sent to the streams channel.
-func portForwardStreamReceived(streams chan httpstream.Stream) func(httpstream.Stream) error {
-	return func(stream httpstream.Stream) error {
+func portForwardStreamReceived(streams chan httpstream.Stream) func(httpstream.Stream, <-chan struct{}) error {
+	return func(stream httpstream.Stream, replySent <-chan struct{}) error {
 		// make sure it has a valid port header
 		portString := stream.Headers().Get(api.PortHeader)
 		if len(portString) == 0 {

--- a/pkg/kubelet/server/server_test.go
+++ b/pkg/kubelet/server/server_test.go
@@ -1699,7 +1699,9 @@ func TestPortForwardStreamReceived(t *testing.T) {
 		if len(test.streamType) > 0 {
 			stream.headers.Set("streamType", test.streamType)
 		}
-		err := f(stream)
+		replySent := make(chan struct{})
+		err := f(stream, replySent)
+		close(replySent)
 		if len(test.expectedError) > 0 {
 			if err == nil {
 				t.Errorf("%s: expected err=%q, but it was nil", name, test.expectedError)

--- a/pkg/util/httpstream/httpstream.go
+++ b/pkg/util/httpstream/httpstream.go
@@ -33,12 +33,12 @@ const (
 
 // NewStreamHandler defines a function that is called when a new Stream is
 // received. If no error is returned, the Stream is accepted; otherwise,
-// the stream is rejected.
-type NewStreamHandler func(Stream) error
+// the stream is rejected. After the reply frame has been sent, replySent is closed.
+type NewStreamHandler func(stream Stream, replySent <-chan struct{}) error
 
 // NoOpNewStreamHandler is a stream handler that accepts a new stream and
 // performs no other logic.
-func NoOpNewStreamHandler(stream Stream) error { return nil }
+func NoOpNewStreamHandler(stream Stream, replySent <-chan struct{}) error { return nil }
 
 // Dialer knows how to open a streaming connection to a server.
 type Dialer interface {

--- a/pkg/util/httpstream/spdy/connection.go
+++ b/pkg/util/httpstream/spdy/connection.go
@@ -120,7 +120,8 @@ func (c *connection) CloseChan() <-chan bool {
 // the stream. If newStreamHandler returns an error, the stream is rejected. If not, the
 // stream is accepted and registered with the connection.
 func (c *connection) newSpdyStream(stream *spdystream.Stream) {
-	err := c.newStreamHandler(stream)
+	replySent := make(chan struct{})
+	err := c.newStreamHandler(stream, replySent)
 	rejectStream := (err != nil)
 	if rejectStream {
 		glog.Warningf("Stream rejected: %v", err)
@@ -130,6 +131,7 @@ func (c *connection) newSpdyStream(stream *spdystream.Stream) {
 
 	c.registerStream(stream)
 	stream.SendReply(http.Header{}, rejectStream)
+	close(replySent)
 }
 
 // SetIdleTimeout sets the amount of time the connection may remain idle before

--- a/pkg/util/httpstream/spdy/roundtripper_test.go
+++ b/pkg/util/httpstream/spdy/roundtripper_test.go
@@ -134,7 +134,7 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 			streamCh := make(chan httpstream.Stream)
 
 			responseUpgrader := NewResponseUpgrader()
-			spdyConn := responseUpgrader.UpgradeResponse(w, req, func(s httpstream.Stream) error {
+			spdyConn := responseUpgrader.UpgradeResponse(w, req, func(s httpstream.Stream, replySent <-chan struct{}) error {
 				streamCh <- s
 				return nil
 			})

